plugin.program.autowidget/resources/lib/refresh.py

```diff
@@ -292,12 +302,16 @@ def cache_and_update(widget_ids):
             if isinstance(path, dict):
                 _label = path['label']
                 path = path['file']['file']
+            else:
+                _label = widget_def.get('group','')
             hash = hashlib.sha1(path).hexdigest()
             # TODO: we might be updating paths used by widgets that weren't initiall queued.
             # We need to return those and ensure they get refreshed also.
             effected_widgets = effected_widgets.union(utils.widgets_for_path(path))
             if utils.is_cache_queue(hash):
                 # we need to update this path regardless
+                if notify:
+                    dialog.notification(u'AutoWidget', u"Updating widget {}".format(_label), sound=False)

```

Comment:
@drinfernoo I'm going to switch this to dialogprogressbg as it would be a lot more helpful. Just have to change how it works so I can estimate the queue length.